Library/Homebrew/formula.rb

```diff
@@ -595,6 +595,14 @@ def inspect
     s << ") #{path}>"
   end
 
+  def file_modified?
+    return false unless which("git")
+    path.parent.cd do
+      diff = Utils.popen_read("git", "diff", "--diff-filter=M", "--name-only", "#{path}")
+      !diff.empty? && $?.exitstatus == 0
+    end
+  end
```

Comment:
Also add this method here: https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/test/test_bottle_hooks.rb#L6-L10